cmd/cmd.go

```diff
@@ -547,61 +547,63 @@ func RecoverConfig(confPath string, defConf *config.Config, chainType genesis.Ch
-	if len(confRewardAddresses) > 1 &&
-		len(confRewardAddresses) != len(valAddrsInfo) {
-		return nil, fmt.Errorf("reward addresses should be %v", len(valAddrsInfo))
-	}
-
-	// Create reward addresses
 	rewardAddrs := make([]crypto.Address, 0, len(valAddrsInfo))
-	if len(confRewardAddresses) != 0 {
-		for _, addrStr := range confRewardAddresses {
-			addr, _ := crypto.AddressFromString(addrStr)
-			rewardAddrs = append(rewardAddrs, addr)
-		}
 
-		if len(rewardAddrs) == 1 {
-			for i := 1; i < len(valAddrsInfo); i++ {
-				rewardAddrs = append(rewardAddrs, rewardAddrs[0])
-			}
-		}
-	} else {
-		for i := 0; i < len(valAddrsInfo); i++ {
-			valAddrPath, _ := addresspath.FromString(valAddrsInfo[i].Path)
-			accAddrPath := addresspath.NewPath(
-				valAddrPath.Purpose(),
-				valAddrPath.CoinType(),
+	switch {
+	// Case 1: No reward addresses in the config file.
+	case len(confRewardAddrs) == 0:
+		// Try to use the first Ed25519 address from the wallet as the reward address.
+		firstEd25519AddrPath := addresspath.NewPath(
+			vault.PurposeBIP44Hardened,
+			wlt.CoinType()+addresspath.HardenedKeyStart,
+			uint32(crypto.AddressTypeEd25519Account)+addresspath.HardenedKeyStart,
+			uint32(0)+addresspath.HardenedKeyStart)
+
+		addrInfo := wlt.AddressFromPath(firstEd25519AddrPath.String())
+		if addrInfo == nil {
+			// If no Ed25519 address is found, try the first BLS address instead.
+			firstBLSAddrPath := addresspath.NewPath(
+				vault.PurposeBLS12381Hardened,
+				wlt.CoinType()+addresspath.HardenedKeyStart,
 				uint32(crypto.AddressTypeBLSAccount)+addresspath.HardenedKeyStart,
-				valAddrPath.AddressIndex())
+				uint32(0))
+
+			addrInfo = wlt.AddressFromPath(firstBLSAddrPath.String())
 
-			addrInfo := walletInstance.AddressFromPath(accAddrPath.String())
 			if addrInfo == nil {
-				accAddrPath = addresspath.NewPath(
-					vault.PurposeBIP44Hardened,
-					valAddrPath.CoinType(),
-					uint32(crypto.AddressTypeEd25519Account)+addresspath.HardenedKeyStart,
-					uint32(0)+addresspath.HardenedKeyStart)
-
-				addrInfo = walletInstance.AddressFromPath(accAddrPath.String())
-				if addrInfo == nil {
-					return nil, fmt.Errorf("unable to find reward address for: %s [%s]",
-						valAddrsInfo[i].Address, accAddrPath)
-				}
+				return nil, fmt.Errorf("unable to find a reward address in the wallet")
 			}
+		}
 
-			addr, _ := crypto.AddressFromString(addrInfo.Address)
+		addr, _ := crypto.AddressFromString(addrInfo.Address)
+		for i := 0; i < len(valAddrsInfo); i++ {
 			rewardAddrs = append(rewardAddrs, addr)
 		}
-	}
 
-	// Check if reward addresses are account address
-	for _, addr := range rewardAddrs {
-		if !addr.IsAccountAddress() {
-			return nil, fmt.Errorf("reward address is not an account address: %s", addr)
+	// Case 2: One reward address is specified in the config file.
+	case len(confRewardAddrs) == 1:
+		// Use this single address for all validators.
+		addr, _ := crypto.AddressFromString(confRewardAddrs[0])
```

Comment:
Why are these kinds of errors never handled? This does not make an impact here but maybe has affect in transfer reward process when this address is not valid.

Reply:
In BasicCheck, we verify the validity of the config file, so by this point, we know the config is valid.